writing-test-plans: Update manifest with new parameters and descriptions (#17)
testground/docs: writing-test-plans/test-plan-manifest.md

# dependencies of the current plan state.
# If this flag is unset or false, every build of a test plan will start
# with a blank go container. If this flag is true, the builder will the last
# cached image.
enable_go_build_cache = false

#  Cgo enables the creation of Go packages that call C code. By default it is disabled.
#  Enabling CGO also enables dynamic linking. Disabling CGO (default) produces statically  linked binaries.
#  If you ever see errors like the following, you are probably better off
#  disabling CGO (and therefore enabling static linking).
#    /testplan: error while loading shared libraries: libdl.so.2: cannot open shared object file: No such file or directory
#  If you pass `true` to this flag, your test plan will be built with CGO_ENABLED=1
enable_cgo = false

[builders."exec:go"]
enabled = true
module_path = "github.com/your/module/name"

[runners."local:docker"]
enabled = true
ulimits = [
  "nofile=1048576:1048576",
]

[daemon]
listen                    = ":8040"

[daemon.scheduler]
task_timeout_min          = 5
task_repo_type            = "disk"

[client]
endpoint = "http://localhost:8040"

> user = "myname"

Reviewer: Can you double-check all these options? It looks like we're passing daemon configuration in a test's manifest.

Author: Yes that was really confusing for me, because I thought that we can change daemon's and engine's configuration from test's manifest because I saw some manifests examples. I would remove all those engine's configuration from test-plan-manifest. What do you think? @laurentsenta

Reviewer: Yes please, could you share the link where you've seen these options used in a test manifest?
This is definitely something we need to remove from wherever it is defined.

Also, in testground: `pkg/api/manifest.go` you'll find the spec for this file this will help filter the correct options.

[runners."cluster:k8s"]
enabled = true

> 

Reviewer: ```suggestion
```

```

### Test cases

Finally, we have [test cases](../concepts-and-architecture/test-structure.md#test-cases). Test cases are defined in an [array of tables](https://github.com/toml-lang/toml#array-of-tables) which specify the name of the test case, boundaries for the number of instances and the values of any parameters being tested in a particular test case.

> Be carefull, if your testplan requires integer number, and you do not provide it, neither the 'default' value, testcase will fail.

Reviewer: ```suggestion
If your plan requires an integer param without a default value, the test will fail if you do not provide it.
```

[[testcases]]
name= "quickstart"
instances = { min = 1, max = 5, default = 1 }

[testcases.params]
  conn_count       = { type = "int", desc = "number of TCP sockets to open" default = 5 }
  conn_outgoing    = { type = "int", desc = "number of outgoing TCP dials", default = 5 }
  conn_delay_ms       = { type = "int", desc = "random milliseconds jitter before TCP dial", default = 30000 }
  concurrent_dials = { type = "int", desc = "max number of concurrent net.Dial calls", default = 10 }

>   data_size_kb     = { type = "int", desc = "size of data to write to each TCP connection", default = 128 }

Reviewer: Are there other `type` options possible? Can we create an option without a default?
Could you replace these very similar examples with one with different types, default, etc? This would give a better view of the options.

Author: Testcase will fail if we do not provide default value. I was using these manifests for testing: https://github.com/testground/learning-example-tg/blob/master/manifest.toml . For example if we do not provide 
 `messages   = { type = "int", default = 5}` 
 we get the error: 
 ```
FAIL << single[001] >> strconv.Atoi: parsing "null": invalid syntax
Sep 23 08:37:43.965515	INFO	0.6839s    MESSAGE << single[000] >> claimed sequence numbers; global=1, group(single)=1
Sep 23 08:37:43.965603	INFO	0.6840s       FAIL << single[000] >> strconv.Atoi: parsing "null": invalid syntax
```

I will add that note to the docs
